src/api/app/controllers/webui/webui_controller.rb

```diff
@@ -169,9 +168,36 @@ def authenticate
     mode = CONFIG['proxy_auth_mode'] || :off
     logger.debug "Authenticating with iChain mode: #{mode}"
     if mode == :on || mode == :simulate
-      authenticate_proxy
+      mode = CONFIG['proxy_auth_mode'] || :off
+      proxy_user = request.env['HTTP_X_USERNAME']
+      proxy_email = request.env['HTTP_X_EMAIL']
+      if mode == :simulate
+        proxy_user ||= CONFIG['proxy_auth_test_user'] || CONFIG['proxy_test_user']
+        proxy_email ||= CONFIG['proxy_auth_test_email']
+      end
+      if proxy_user
+        session[:login] = proxy_user
+        session[:email] = proxy_email
+        ActiveXML.api.delete_additional_header 'X-Username'
+        ActiveXML.api.delete_additional_header 'X-Email'
+        ActiveXML.api.delete_additional_header 'Authorization'
+        # Set the headers for direct connection to the api, TODO: is this thread safe?
+        ActiveXML.api.set_additional_header( 'X-Username', proxy_user )
+        ActiveXML.api.set_additional_header( 'X-Email', proxy_email ) if proxy_email
```

Comment:
Line is too long. [84/80]<br>Space inside parentheses detected.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.